Commands/AnnouncementCmds.cs

```diff
@@ -265,41 +268,50 @@ public async Task AnnounceBuildSlashCommand(SlashCommandContext ctx,
             }
         }
 
-        [Command("editannouncetextcmd")]
-        [TextAlias("editannounce")]
+        [Command("editannounce")]
         [Description("Edit an announcement, preserving the ping highlight.")]
-        [AllowedProcessors(typeof(TextCommandProcessor))]
+        [AllowedProcessors(typeof(SlashCommandProcessor))]
         [RequireHomeserverPerm(ServerPermLevel.Moderator)]
         public async Task EditAnnounce(
-            TextCommandContext ctx,
-            [Description("The ID of the message to edit.")] ulong messageId,
-            [Description("The short name for the role to ping.")] string roleName,
-            [RemainingText, Description("The new message content, excluding the ping.")] string content
+            SlashCommandContext ctx,
+            [Parameter("message"), Description("The ID of the message to edit.")] string messageId,
+            [Parameter("role1"), Description("The short name for the first role to ping.")] string role1Name,
+            [Parameter("role2"), Description("The short name for the second role to ping. Optional.")] string role2Name = null
```

Comment:
These should be a choice list, similar to `WindowsInsiderChannelChoiceProvider` but with the full list of roles pulled from config. I wonder if you can just iterate the config and add `new()` entries dynamically?

https://github.com/Erisa/Cliptok/blob/4011df6bbd7cc609073bf4fffb208cac5928c4ea/Commands/AnnouncementCmds.cs#L434C24-L447